core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -34,20 +33,16 @@ use {
 };
 
 /// Controls packet and transaction flow into scheduler, and scheduling execution.
-pub(crate) struct SchedulerController<
-    C: LikeClusterInfo,
-    Tx: TransactionWithMeta,
-    R: ReceiveAndBuffer<Tx>,
-> {
```

Comment:
Idk I prefer keeping the `Tx` generic parameter personally. Feels backwards to be getting it from the `ReceiveAndBuffer` rather than passing it

Reply:
Idea I had here is that `ReceiveAndBuffer`'s type(s) are a single-point controlling all the inner implementations.

I also think this sort of set up is somewhat necessary with what I have planned.
(See most recent push, `Container` type is also chosen by the `ReceiveAndBuffer`).

Old transaction type:
- receive packet batches, deserialize into `ImmutableDeserializedPacket`
- do filtering, eventually end up with `RuntimeTransaction<SanitizedTransaction>`
- insert those into the `TransactionStateContainer<RuntimeTransaction<SanitizedTransaction>>`

New transaction type:
- receive packet batches. get **pre-allocated** space `(BytesMut)` from the container, copy bytes of packet, parse (using `Bytes`).
- do filtering, eventually end up with `RuntimeTransaction<ResolvedTransactionView>`
- insert those into the the a different container type (`WrappedTransactionStateContainer`).
  - this wrapping is necessary so that we can get and re-use the `Bytes` for the transactions

So it doesn't make sense if we have the `TransactionViewReceiveAndBuffer` and a `TransactionStateContainer<..>`, because we need slightly different behavior due to the type differences; and they require a slightly different interface as well, since getting free space to work on received packets isn't something the base `TransactionStateContainer` needs to do.


A lot of this gets simplified once we are receiving `Bytes` from upstream, because we could just insert those into the `TransactionStateContainer`. But that's blocked rn because of CUDA stuff (that no one uses) 🫠

Reply:
^ kind of a long winded explanation. let me know if you want to hop on a call to discuss, or if you have questions here it's fine too!